force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,54 @@
+trigger OpportunityTrigger on Opportunity (before update, before delete) {
+ /*
+    * Question 5
+    * Opportunity Trigger
+    * When an opportunity is updated validate that the amount is greater than 5000.
+    * Error Message: 'Opportunity amount must be greater than 5000'
+    * Trigger should only fire on update.
+    */
+    /*
+    * Question 7
+    * Opportunity Trigger
+    * When an opportunity is updated set the primary contact on the opportunity to the contact on the same account with the title of 'CEO'.
+    * Trigger should only fire on update.
+    */
+    if(Trigger.isBefore && Trigger.isUpdate){
+        for(Opportunity opp : Trigger.new){
+            if(opp.Amount < 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+        Set<Id> accIds = new Set<Id>();
+        for(Opportunity opp : Trigger.new){
+            accIds.add(opp.AccountId);
+        }
```

Comment:
Can you also gather this information what you are iterating for the addError? Reduce the amount of times your iterating trigger.new